src/pages/Swap/SwapButton.tsx

```diff
@@ -0,0 +1,233 @@
+import { Trans } from '@lingui/macro'
+import { sendAnalyticsEvent, TraceEvent, useTrace } from '@uniswap/analytics'
+import { BrowserEvent, InterfaceElementName, InterfaceEventName } from '@uniswap/analytics-events'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { UNIVERSAL_ROUTER_ADDRESS } from '@uniswap/universal-router-sdk'
+import { useWeb3React } from '@web3-react/core'
+import { useToggleAccountDrawer } from 'components/AccountDrawer'
+import { ButtonError, ButtonLight, ButtonPrimary } from 'components/Button'
+import { GrayCard } from 'components/Card'
+import { MouseoverTooltip } from 'components/Tooltip'
+import { getChainInfo } from 'constants/chainInfo'
+import { isSupportedChain, SupportedChainId } from 'constants/chains'
+import usePermit2Allowance, { AllowanceState } from 'hooks/usePermit2Allowance'
+import useWrapCallback, { WrapType } from 'hooks/useWrapCallback'
+import JSBI from 'jsbi'
+import { useCallback, useContext, useMemo, useState } from 'react'
+import { Info, Loader } from 'react-feather'
+import { TradeState } from 'state/routing/types'
+import { DerivedSwapInfoContext } from 'state/swap/hooks'
+import { ThemedText } from 'theme'
+import invariant from 'tiny-invariant'
+import { getIsValidSwapQuote } from 'utils/getIsValidSwapQuote'
+import { switchChain } from 'utils/switchChain'
+
+import { Field } from '../../state/swap/actions'
+import { WrapButton } from './WrapButton'
+
+interface SwapButtonProps {
+  chainId: SupportedChainId | undefined
+  onSwapClick: () => void
```

Comment:
why pass `onSwapClick` as a prop when we can derive it from hooks like the Wrap callback below - maybe remove this